torch/lib/c10d/ProcessGroupMPI.cpp

```diff
@@ -815,8 +832,8 @@ c10::intrusive_ptr<ProcessGroup::Work> ProcessGroupMPI::send(
   }
 
   return c10::make_intrusive<AsyncWork>(
-      tensor,
       request,
+      nullptr,
```

Comment:
hmm, any reason that output tensors is of type `std::vector<at::Tensor>*` while inputTensors is of type `optional<std::vector<at::Tensor>>`?

Reply:
> we don't need to copy outputs here as user of send/recv manages lifetime of tensors

does this mean inputs' lifetime is managed differently, so that we need to copy inputs?

Reply:
I think we do not specify this in our API directly.

I've updated `dst` management: since none of the ops directly modify it, we can make it const and hence WorkMPI and WorkEntry can each have their own copy of outputTensors_.